includes/discount-functions.php

```diff
@@ -32,19 +32,29 @@ function edd_get_discounts( $args = array() ) {
 
 	$args = wp_parse_args( $args, $defaults );
 
-	$discounts = get_posts( $args );
+	$discounts_hash = md5( json_encode( $args ) );
+	if ( ! $discounts = wp_cache_get( $discounts_hash, 'edd-discounts' ) ) {
+		$discounts = get_posts( $args );
+		wp_cache_set( $discounts_hash, $discounts, 'edd-discounts' );
```

Comment:
An expiry needs to be set to `3600` otherwise it default to `0` which means no expiration.